### PR TITLE
Go module migration 

### PR DESCRIPTION
**Pull request by vaniisgh**
#126 
TO-DO:
 - fix-ci
- test builds

### CODE REVIEW

**Comment by pawanpraka1** [comment]
@prateekpandey14 can you take a look, it is still in draft because of the issues. Can you help resolving those issues?

**Comment by pawanpraka1** [comment]
@vaniisgh, why you are changing the travis.yaml?

**Comment by vaniisgh** [comment]
> @vaniisgh, why you are changing the travis.yaml?

hey so, I run an OSX system and didn't want to run a VirtualBox VM to check my changes so I thought I could do the same by making some fixes in the Travis ci file and test them that way.
But the ci will fail for sure because of this part:
```
install:
  - if [ "$TRAVIS_BUILD_DIR" != "$GOPATH/src/github.com/openebs/zfs-localpv" ]; then
    mkdir -p $GOPATH/src/github.com/openebs/;
    mv $TRAVIS_BUILD_DIR $GOPATH/src/github.com/openebs;
    cd $GOPATH/src/github.com/openebs/zfs-localpv;
```
since the plan is to switch to gomodules, this wont work.

not sure I am right but that was my reasoning :) 
thanks again :)

File: .travis.yml [hunk 1]
```diff
@@ -23,11 +23,10 @@ before_install:
...
 install:
-  - if [ "$TRAVIS_BUILD_DIR" != "$GOPATH/src/github.com/openebs/zfs-localpv" ]; then
-    mkdir -p $GOPATH/src/github.com/openebs/;
-    mv $TRAVIS_BUILD_DIR $GOPATH/src/github.com/openebs;
-    cd $GOPATH/src/github.com/openebs/zfs-localpv;
-    fi
+  - if [ "$TRAVIS_BUILD_DIR" != "$HOME/openebs/zfs-localpv" ]; then
+    mkdir -p $HOME/openebs/;
+    mv $TRAVIS_BUILD_DIR $HOME/openebs;
+    cd $HOME/openebs/zfs-localpv;
```
Comment (prateekpandey14) [inline .travis.yml:29]: i think we don't need this anymore , we can remove this , travis handles the gopath correctly now.

Comment (vaniisgh) [reply .travis.yml:29]: can I just remove it completely then?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	google.golang.org/grpc v1.27.0
+	google.golang.org/protobuf v1.24.0 // indirect
+	k8s.io/api v0.0.0
+	k8s.io/apimachinery v0.0.0
+	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
+	k8s.io/klog v0.4.0
+	k8s.io/kube-openapi v0.0.0-20190816220812-743ec37842bf // indirect
+	k8s.io/kubernetes v1.15.3
+	k8s.io/utils v0.0.0-20190829053155-3a4a5477acf8 // indirect
+	sigs.k8s.io/controller-runtime v0.2.0
```
Comment (prateekpandey14) [inline go.mod:40]: we are not using controller-runtime , we can remove this

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	k8s.io/cri-api => k8s.io/cri-api v0.0.0-20190531030430-6117653b35f1
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.0.0-20190620090116-299a7b270edc
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20190620085325-f29e2b4a4f84
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20190620085942-b7f18460b210
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20190620085809-589f994ddf7f
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20190620085912-4acac5405ec6
+	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20190620085838-f1cb295a73c9
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20190620090156-2138f2c9de18
+	k8s.io/metrics => k8s.io/metrics v0.0.0-20190620085625-3b22d835f165
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20190620085408-1aef9010884e
```
Comment (prateekpandey14) [inline go.mod:63]: same as above comment , sample-apiserver is not getting used, we can remove this as well.
Run the command `go mod tidy & go mod verify` to remove some of unused dependencies.

File: .travis.yml (hunk 1 above)
Comment (prateekpandey14) [reply .travis.yml:29]: yes, please

File: go.mod (hunk 3 above)
Comment (vaniisgh) [reply go.mod:63]: I do run a tidy, I run in the makefile as well.
should I include a verify too?

Comment (vaniisgh) [reply go.mod:63]: also, it gives me the same issue as the Maya repo. Heres the link to the issue that made me think I needed to add all these dependencies: https://github.com/kubernetes/kubernetes/issues/90358

**Pull request closed without merging**